### ISSUE TITLE
Is the API needed?

### ISSUE THREAD

**Issue by sixcorners** [Oceanhips/Destiny-Debate-Hub#6]
Would it make sense to just include the json files in the git repo and import them? It's doesn't seem like a whole lot of data. I think the import statement works on json files. If this was just a static site you could probably run this on cheaper hosting or deploy it onto ipfs.

**Comment by Oceanhips** [comment]
Might be misunderstanding the question but hopefully my response makes sense.  Basically there are a couple of reasons for this choice but it's definitely debatable depending on the situation.

The biggest advantage for creating an API that hosts all the DB data instead of building out raw JSON then importing the files is that I can adjust and make changes to the database without ever having to redeploy the frontend or backend each time when I make a change.  This allows me to make very quick adjustments (i.e. I spelled a tag wrong or used the wrong link for a YT video) and then just update the DB.  Once updated, the only thing that has to be done is just refresh the page; the API makes a GET request to get the data and returns the updated data which is then dynamically populated on the page.

If I have the local JSON stored in the repo (in the backend presumably) I would have to push up the commits to the repo then going into the Linux server that the site is being hosted on, pull down the commits from the master (CI/CD could technically help automate part of this process) and then restart the server.  This process would have to be done in its entirety for any change made to the JSON data. 

For smaller, more static data this isn't a huge issue (A good example of this is the press page.  Since that data rarely changes I just hard coded it out instead of using an API) however for the debate data since I add ~2 debates a day roughly; it would dramatically increase the amount of time spend adjusting/redeploying everything.

Basically this method allows me to never had to touch the frontend or backend at all while still being able to update and add debates to the site dynamically.

Lastly because of the size of the site's of the database I've paid a total of 0$ to AWS to host the data and API so at least for now this method costs me nothing.

**Comment by sixcorners** [comment]
What I mean by cheaper hosting is you can use S3 or B2 or netlify and omit the VM running nginx. (You could probably do that now actually even without making changes with how the data is stored.) If it was in the repo it would be version controlled and people would be able to submit suggested changes. When I pull the code it would have worked instead of erroring out. CI/CD is kind of a good thing to have even if you keep using a database. You could kind of say that doing this would make CI/CD a "must have" and that it would help get that done. I don't think doing something like this is supposed to dramatically change the level of effort spent anywhere once automated deployments/tests are in place. There would be less moving parts and stuff that can break at least once it's deployed. It would be faster, though it's already pretty fast, because everything would be cached on the client. It would allow it to work offline, though that really doesn't matter here since the videos don't work offline. You could expose it over IPFS.

I think the main downside is that you wouldn't be able to give users an interface to make changes directly on the site in the future.

I guess I just read https://18f.gsa.gov/2015/03/19/how-we-built-analytics-usa-gov/ a while ago and wonder now how every website can be built like that. Also I guess I've been burned by the reliability of https://zippopotam.us/ which is an apache/wsgi/python/mongodb thing that I think could maybe use the same strategy as that blog post.

Anyway, it's just a thought. Databases are great too. Thanks for making the website.
